src/Repository/OAuth/RefreshTokenRepository.php

```diff
@@ -46,12 +47,39 @@ public function findAllRefreshTokensByClient(Client $client): array
         ;
     }
 
+    /**
+     * @return RefreshToken[]
+     */
+    public function findAllRefreshTokensByUser(Adherent $user): array
+    {
+        return $this
+            ->createQueryBuilder('rt')
+            ->join('rt.accessToken', 'at')
+            ->where('at.user = :user')
```

Comment:
what about adding this check `AND rt.revokedAt IS NULL` to avoid to take already revoked tokens? same for accessToken query